engine/particle/src/particle.cpp

```diff
@@ -1016,12 +1030,79 @@ namespace dmParticle
         particle->m_SourceAngularVelocity = emitter_properties[EMITTER_KEY_PARTICLE_ANGULAR_VELOCITY];
     }
 
-    static float unit_tex_coords[] =
+    static inline bool HasLocalPositionAttribute(const ParticleVertexAttributeInfos& attribute_infos)
     {
-            0.0f,1.0f, 0.0f,0.0f, 1.0f,0.0f, 1.0f,1.0f
+        for (int i = 0; i < attribute_infos.m_NumInfos; ++i)
+        {
+            if (attribute_infos.m_Infos[i].m_SemanticType == dmGraphics::VertexAttribute::SEMANTIC_TYPE_POSITION &&
+                attribute_infos.m_Infos[i].m_CoordinateSpace == dmGraphics::COORDINATE_SPACE_LOCAL)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
A small performance thing, we can skip calculating local positions in the update function if we don't need it.